src/main/java/com/iopipe/plugin/logger/LoggerExecution.java

```diff
@@ -0,0 +1,543 @@
+package com.iopipe.plugin.logger;
+
+import com.iopipe.http.RemoteException;
+import com.iopipe.IOpipeExecution;
+import com.iopipe.plugin.IOpipePlugin;
+import com.iopipe.plugin.IOpipePluginExecution;
+import com.iopipe.plugin.IOpipePluginPostExecutable;
+import com.iopipe.IOpipeSigner;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.CharBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import javax.json.Json;
+import javax.json.JsonObject;
+import org.pmw.tinylog.Logger;
+
+/**
+ * Contains the execution state for the logger plugin.
+ *
+ * @since 2018/09/24
+ */
+public final class LoggerExecution
+	implements IOpipePluginExecution
+{
+	/** Lock on logging to prevent spliced logs. */
+	protected final Object lock =
+		new Object();;
+	
+	/** The temporary file. */
+	protected final Path tempfile;
+	
+	/** The channel for the data. */
+	protected final FileChannel channel;
+	
+	/** The stream to write to for JSON data. */
+	protected final Writer writer;
+	
+	/** The signer. */
+	private final IOpipeSigner _signer;
+	
+	/**
+	 * Initializes the logger plugin collector.
+	 *
+	 * @param __exec The execution.
+	 * @throws NullPointerException On null arguments.
+	 * @since 2018/09/24
+	 */
+	public LoggerExecution(IOpipeExecution __exec)
+		throws NullPointerException
+	{
+		if (__exec == null)
+			throw new NullPointerException();
+		
+		IOpipeSigner signer = __exec.signer(".log");
+		
+		// The signer might not be available or the logging plugging parts
+		// might not initialize
+		Path tempfile = null;
+		FileChannel channel = null;
+		Writer writer = null;
+		
+		// If the signer is available, setup the log to print to
+		if (signer != null)
+			try
+			{
+				// Store log data in a temporary file
+				tempfile = Files.createTempFile("iopipe-logger", ".log");
+				
+				// Open temporary file for read/write
+				channel = FileChannel.open(tempfile,
+					StandardOpenOption.DELETE_ON_CLOSE,
+					StandardOpenOption.CREATE,
+					StandardOpenOption.TRUNCATE_EXISTING,
+					StandardOpenOption.READ,
+					StandardOpenOption.WRITE);
+				
+				writer = new OutputStreamWriter(
+					Channels.newOutputStream(channel), "utf-8");
+			}
+			catch (IOException e)
+			{
```

Comment:
Consider doing some kind of logging here, how are you going to catch issues if they arise?

Reply:
After discussing, we'll hold on adding logging here out of recursive concerns -- if/when needs arise, we'll revisit where we need to add observability in here